src/session.rs

```diff
@@ -327,19 +344,34 @@ impl Session {
             }
         }
     }
+    #[cfg(feature = "openssl-tls")]
+    pub(crate) async fn new_openssl(
+        webdriver: &str,
+        cap: webdriver::capabilities::Capabilities,
+    ) -> Result<Client, error::NewSessionError> {
+        let connector = hyper_tls::HttpsConnector::new();
+        Self::with_capabilities(webdriver, cap, connector).await
+    }
+    pub(crate) async fn new_rustls(
+        webdriver: &str,
+        cap: webdriver::capabilities::Capabilities,
+    ) -> Result<Client, error::NewSessionError> {
+        let connector = hyper_rustls::HttpsConnector::new();
+        Self::with_capabilities(webdriver, cap, connector).await
+    }
```

Comment:
These `impl`s probably need to be in their own `impl` blocks for the specific `Client<C>` that they each produce.